daemon/cmd/endpoint.go

```diff
@@ -421,37 +436,68 @@ func (d *Daemon) createEndpoint(ctx context.Context, owner regeneration.Owner, e
 		}
 	}
 
+	identityLbls := maps.Clone(apiLabels)
+
 	var cancel context.CancelFunc
 	ctx, cancel = context.WithCancel(ctx)
 	d.endpointCreations.NewCreateRequest(ep, cancel)
-	defer d.endpointCreations.EndCreateRequest(ep)
-
-	identityLbls := maps.Clone(apiLabels)
-
 	if ep.K8sNamespaceAndPodNameIsSet() && d.clientset.IsEnabled() {
-		pod, cp, k8sIdentityLbls, k8sInfoLbls, annotations, err := d.fetchK8sMetadataForEndpoint(ep.K8sNamespace, ep.K8sPodName)
+		pod, k8sMetadata, err := d.fetchK8sMetadataForEndpoint(ep.K8sNamespace, ep.K8sPodName)
+		if pod != nil && ep.K8sUID != string(pod.GetUID()) {
+			log.WithFields(logrus.Fields{
+				logfields.K8sPodName: ep.K8sNamespace + "/" + ep.K8sPodName,
+				logfields.K8sUID:     ep.K8sUID,
+			}).Warn("Detected outdated Pod UID during Endpoint creation. Endpoint creation cannot proceed with an outdated Pod store. Creation will be retried.")
+
+			t, ok := d.endpointCreations.requests[d.endpointCreations.getKey(ep)]
+			if !ok || time.Now().Sub(t.started) < 30*time.Second {
+				return ep, 0, fmt.Errorf("pod store out-of-date")
```

Comment:
So we return failure here, and wait for kubelet -> containerd -> cni -> agent to retry? Am I reading this right?

Does it make sense to pause for a few seconds first to let the informer catch up? That would save some unnecessary system load.

Reply:
Ok, I've pushed this suggestion, PTAL.